backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/CertificationValidityChecker.java

```diff
@@ -70,15 +71,24 @@ private void checkCertificationValidity() {
                     AuditLogType.ENGINE_CA_CERTIFICATION_HAS_EXPIRED,
                     AuditLogType.ENGINE_CA_CERTIFICATION_IS_ABOUT_TO_EXPIRE_ALERT,
                     AuditLogType.ENGINE_CA_CERTIFICATION_IS_ABOUT_TO_EXPIRE,
-                    null)
-                    ^ !checkCertificate((X509Certificate) EngineEncryptionUtils.getCertificate(),
-                    AuditLogType.ENGINE_CERTIFICATION_HAS_EXPIRED,
-                    AuditLogType.ENGINE_CERTIFICATION_IS_ABOUT_TO_EXPIRE_ALERT,
-                    AuditLogType.ENGINE_CERTIFICATION_IS_ABOUT_TO_EXPIRE,
                     null)) {
+                // if engine CA is going to expire, then it needs to be renewed first and only then it makes sense
+                // to renew other certificates
                 return;
             }
 
+            checkCertificate((X509Certificate) EngineEncryptionUtils.getCertificate(),
+                    AuditLogType.ENGINE_CERTIFICATION_HAS_EXPIRED,
+                    AuditLogType.ENGINE_CERTIFICATION_IS_ABOUT_TO_EXPIRE_ALERT,
+                    AuditLogType.ENGINE_CERTIFICATION_IS_ABOUT_TO_EXPIRE,
+                    null);
```

Comment:
In case the Engine certificate is already expired, maybe checking the host certificates below will fail.  But exceptions look handled here, so the fact that we don't return after this check anymore shouldn't cause a real problem (and it's probably more correct because now we can warn about host certificates when the Engine certificate is going to expire).

Reply:
AFAIR VDSM is not performing client certificate checks, so even if the engine certificate is expired, engine would still be able to connect to hosts. So host certificates check should fail only when host certificate expires or going to expire.